modules/workergroup/nodepools.tf

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017, 2022, Oracle Corporation and/or affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+# Dynamic resource block for Node Pool groups defined in worker_groups
+resource "oci_containerengine_node_pool" "nodepools" {
+  for_each = {
+    for i in local.worker_groups : i.name => i
+    if coalesce(lookup(i, "mode", "unknown"), local.mode) == "node-pool"
+    && lookup(i, "create", var.create)
+  }
+
+  compartment_id = lookup(each.value, "compartment_id", local.compartment_id)
+  cluster_id     = var.cluster_id # TODO Data source for e.g. oci_containerengine_cluster.k8s_cluster.id
+
+  kubernetes_version = local.cluster_k8s_version
+  name               = join("-", compact([lookup(each.value, "label_prefix", var.label_prefix), each.key]))
+  freeform_tags      = merge(coalesce(local.freeform_tags, {}), lookup(each.value, "freeform_tags", { oke_group = each.key }))
+  defined_tags       = merge(coalesce(local.defined_tags, {}), lookup(each.value, "defined_tags", {}))
+
+  node_config_details {
+    size = max(0, lookup(each.value, "size", local.size))
+
+    is_pv_encryption_in_transit_enabled = var.enable_pv_encryption_in_transit
+    kms_key_id                          = var.volume_kms_key_id
+
+    dynamic "placement_configs" {
+      iterator = ad_iterator
+      for_each = [for n in lookup(each.value, "placement_ads", local.ad_numbers) :
+        length(local.ad_numbers) == 1 ? local.ad_number_to_name[1] : local.ad_number_to_name[n]
+      ]
+      content {
+        availability_domain = ad_iterator.value
+        subnet_id           = var.primary_subnet_id
+      }
+    }
+
+    nsg_ids = lookup(each.value, "worker_nsg_ids", var.worker_nsg_ids)
+
+    # flannel requires cni type only
+    dynamic "node_pool_pod_network_option_details" {
+      for_each = coalesce(var.cni_type, "flannel") == "flannel" ? [1] : []
```

Comment:
In the default node pools, the code to check for cni_type is

```for_each = var.cni_type == "flannel" ? [1] : []```

Can we please make it standard?